source/game/system/CourseMap.cc

```diff
@@ -180,6 +185,26 @@ f32 CourseMap::getCheckPointEntryOffsetExact(u16 i, const EGG::Vector3f &pos,
     return checkPoint->getEntryOffsetExact(prevPos_, pos_);
 }
 
+s16 CourseMap::getCurrentAreaID(s16 i, const EGG::Vector3f &pos, MapdataAreaBase::Type type) const {
+    // Check if we're colliding with the provided area ID
+    if (i >= 0) {
+        const MapdataAreaBase *area = getArea(i);
+        if (area->type() == type && area->test(pos)) {
+            return i;
+        }
+    }
+
+    // Search all areas of the same type
+    for (i = 0; i < getAreaCount(); ++i) {
```

Comment:
Wy would this for loop be required? Will this function be called with the wrong index `i`? Just curious if you know

Reply:
The index will expire when the point moves outside the area's range. The for loop is a backup option in this event.

Suppose area A is a subset of area B and is sorted later than area B. We will only ever find area B with the default behavior of only having this for loop, so we also cannot remove the priority check.